src/main/java/dk/kb/util/other/StringListUtils.java

```diff
@@ -0,0 +1,281 @@
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class StringListUtils {
+    
+    
+    public static final Predicate<String> notNull = Predicate.not(Objects::isNull);
+    public static final Predicate<String> notEmptyString = Predicate.not(String::isEmpty);
+    public static final Predicate<String> isNotEmpty = notNull.and(notEmptyString);
+    
+    /**
+     * Turn an iterator into a stream
+     *
+     * @param iterator the iterator
+     * @param <T>      the the type of elements
+     * @return A stream of the iterator elements in the same order
+     */
+    public static <T> Stream<T> asStream(Iterator<T> iterator) {
+        return StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED),
+                false);
+    }
+    
+    /**
+     * Return the string or "" if null
+     *
+     * @param string the string
+     * @return same or "" if null
+     */
+    public static String notNull(String string) {
+        return Optional.ofNullable(string).orElse("");
+    }
+    
+    /**
+     * Return the first entry in the list or defaultValue if the list is empty
+     *
+     * @param list         the list
+     * @param defaultValue the value if the list is empty
+     * @return first value or default value if list is empty
+     */
+    public static String getFirst(List<String> list, String defaultValue) {
+        if (list == null || list.isEmpty()) {
+            return defaultValue;
+        } else {
+            return list.get(0);
+        }
+    }
+    
+    /**
+     * If param is not null return that. Otherwise return the value from props[propKey]
+     *
+     * @param props   the properties to default to
+     * @param param   the value to return if notNull
+     * @param propKey the key to props if param is null
+     * @return param or props[propKey]
+     */
+    public static String getParamOrProps(Properties props, String param, String propKey) {
+        if (param == null || param.isBlank()) {
+            
+            if (!props.contains(propKey)) {
+                param = props.getProperty(propKey);
+            } else {
+                param = null;
+            }
+        }
+        return param;
+    }
+    
+    /**
+     * Remove empty elements from the list and distinct the list.
+     *
+     * if list is null, return an empty list.
+     *
+     * @param list the list to filter
+     * @return a list without any empty elements
+     */
+    public static List<String> removeEmpties(List<String> list) {
+        if (list == null) {
+            return Collections.emptyList();
+        }
+        return list.stream()
+                   .filter(Objects::nonNull)
+                   .filter(element -> !element.trim().isEmpty())
+                   .distinct()
+                   .collect(Collectors.toList());
+    }
+    
+    /**
+     * Remove empty elements from the list and distinct the list.
```

Comment:
The JavaDoc should state that the order of the elements is not preserved.

Reply:
The order is preserved, but elements are removed

Reply:
Ah yes, the stream is sequential. My bad.